HealthcareApp/src/Healthcare.Infrastructure/Repository/TokenGeneratorService.cs

```diff
@@ -0,0 +1,54 @@
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using Application.Abstractions;
+using Application.DTOs.Login;
+using Healthcare.Infrastructure.Persistance;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Infrastructure.Repository;
+
+public class TokenGeneratorService : ITokenGeneratorService
+{
+    private readonly UserManager<ApplicationUser> _userManager;
+    private readonly IConfiguration _configuration;
+
+    public TokenGeneratorService(UserManager<ApplicationUser> userManager, IConfiguration configuration)
+    {
+        _userManager = userManager;
+        _configuration = configuration;
+    }
+
+    public async Task<string> GenerateJwtToken(LoginUserDTO loginUserDto)
```

Comment:
You'd better to accept ApplicationUser with list of roles in a property, or 2 parameters - ApplicationUser and List<Role>. Because it's not convenient to use this DTO for token generation, you have to call database each time. And moreover, your class has more dependencies, such as UserManager which is not mandatory for this.
In other words, you can pass user from parameters, but not search it in your method. This will look more logical and clean